test/Chainweb/Test/Pact/RemotePactTest.hs

```diff
@@ -215,32 +216,38 @@ sendValidationTest iot nio =
         h _ = return Nothing
 
 
-spvRequests :: IO (Time Integer) -> IO ChainwebNetwork -> TestTree
-spvRequests iot nio = testCaseSteps "spv client tests" $ \step -> do
+spv :: IO (Time Integer) -> IO ChainwebNetwork -> TestTree
+spv iot nio = testCaseSteps "spv client tests" $ \step -> do
     cenv <- fmap _getClientEnv nio
     batch <- mkTxBatch
     sid <- mkChainId version (0 :: Int)
     r <- flip runClientM cenv $ do
 
       void $ liftIO $ step "sendApiClient: submit batch"
-      rks <- sendApiCmd (testCmdsChainId sid) batch
+      rks <- liftIO $ sending (testCmdsChainId sid) cenv batch
 
       void $ liftIO $ step "pollApiClient: poll until key is found"
-      void $ liftIO $ pollWithRetry (testCmdsChainId sid) cenv rks
+      void $ liftIO $ polling (testCmdsChainId sid) cenv rks
 
       liftIO $ sleep 6
 
       void $ liftIO $ step "spvApiClient: submit request key"
-      pactSpvApiClient version sid (go rks)
+      liftIO $ recoverAll (exponentialBackoff 10000 <> limitRetries 15) $ \s -> do
```

Comment:
What is the maximum wait time here? `10000 * 2^15`? that's about 5 minutes. That seems far to long for a unit test.

Reply:
You may use `step` to log retries (in a way that integrates with tasty's test reporting).

Reply:
Moved to 10, so that it's at max a 10 second wait (10000 * 2^10 ~ 10 s). I've opted to use `debug` to log what we need, since we don't need this data all of the time.